controllers/propagator/propagation.go

```diff
@@ -945,6 +993,31 @@ func (r *PolicyReconciler) processTemplates(
 		}
 
 		policyT.ObjectDefinition.Raw = resolveddata
+
+		// Set initialization vector annotation on the ObjectDefinition for the controller's use
+		if usesEncryption {
+			policyTObjectUnstructured := &unstructured.Unstructured{}
+
+			jsonErr := json.Unmarshal(resolveddata, policyTObjectUnstructured)
+			if jsonErr != nil {
+				return fmt.Errorf("failed to unmarshal the object definition to JSON: %w", jsonErr)
+			}
+
+			policyTAnnotations := policyTObjectUnstructured.GetAnnotations()
+			if policyTAnnotations == nil {
+				policyTAnnotations = make(map[string]string)
+			}
+
+			policyTAnnotations[ivAnnotation] = annotations[ivAnnotation]
+			policyTObjectUnstructured.SetAnnotations(policyTAnnotations)
+
+			updatedPolicyT, jsonErr := json.Marshal(policyTObjectUnstructured)
```

Comment:
Optional: You could avoid the JSON marshaling if the IV annotation is already correctly set.

Reply:
I like it 🙂